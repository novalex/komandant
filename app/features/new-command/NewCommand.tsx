import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import styles from './style.css';
import { selectFields } from './newCommandSlice';

export default function NewCommand() {
	const dispatch = useDispatch();
	const value = useSelector(selectFields);
	return (
		<div>
			<div className={`counter ${styles.counter}`} data-tid="counter">
				{value}
			</div>
		</div>
	);
}
