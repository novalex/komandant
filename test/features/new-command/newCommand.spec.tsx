import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';

import NewCommand from '../../../app/features/new-command';
import reducer, { initialState } from '../../../app/features/new-command/slice';

Enzyme.configure({ adapter: new Adapter() });

function setup(
	preloadedState: { newCommand: typeof initialState } = {
		newCommand: initialState,
	}
) {
	const store = configureStore({
		reducer: { newCommand: reducer },
		preloadedState,
	});

	const getWrapper = () =>
		mount(
			<Provider store={store}>
				<Router>
					<NewCommand />
				</Router>
			</Provider>
		);
	const component = getWrapper();
	return {
		store,
		component,
	};
}

describe('newCommand component', () => {
	it('should match exact snapshot', () => {
		const { store } = setup();
		const tree = renderer
			.create(
				<Provider store={store}>
					<Router>
						<NewCommand />
					</Router>
				</Provider>
			)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
