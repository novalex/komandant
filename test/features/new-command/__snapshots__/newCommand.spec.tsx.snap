// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newCommand component should match exact snapshot 1`] = `
<div
  className="newCommandWrap"
  data-tid="new-command"
>
  <div
    className="field"
  >
    <label
      htmlFor="command-name"
    >
      Name
    </label>
    <input
      id="command-name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="field"
  >
    <label
      htmlFor="command-content"
    >
      Command
    </label>
    <textarea
      id="command-content"
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    className="field"
  >
    <div
      className="button save-command"
      data-tid="save-command"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        
        <span>
          Save
        </span>
      </button>
    </div>
  </div>
</div>
`;
